src/Http/RequestFactory.php

```diff
@@ -218,4 +231,38 @@ public function createHttpRequest()
 		return new Request($url, NULL, $post, $files, $cookies, $headers, $method, $remoteAddr, $remoteHost, $rawBodyCallback);
 	}
 
+	/**
+	 * @param bool $usingTrustedProxy
+	 *
+	 * @return string 'http' or 'https'
+	 */
+	private function getScheme($usingTrustedProxy)
+	{
+		if ($usingTrustedProxy) {
+			if (!empty($_SERVER['HTTP_X_FORWARDED_PROTO'])) {
+				if (strcasecmp($_SERVER['HTTP_X_FORWARDED_PROTO'], 'https') === 0) {
+					return 'https';
+				}
+
+				return 'http';
+			}
+
+			if (!empty($_SERVER['HTTP_X_FORWARDED_PORT'])) {
+				$port = (int) $_SERVER['HTTP_X_FORWARDED_PORT'];
+
+				if ($port === 443) {
+					return 'https';
+				} elseif ($port === 80) {
+					return 'http';
+				}
+			}
```

Comment:
I think it's the same case as with the conflicting scheme and ports. If we decide that we don't want the code to be that clever than we can remove this as well. It only handles badly configured proxies anyway.